File: packages/java/src/main/java/com/tesote/sdk/errors/InvalidOrderStateException.java
package com.tesote.sdk.errors;

/**
 * 409 — order is not in a state that permits the attempted transition
 * (e.g., submitting a non-draft order, cancelling a completed order).
 */
public final class InvalidOrderStateException extends ApiException {
    public InvalidOrderStateException(String message, String errorCode, int httpStatus,
                                      String requestId, String errorId, Integer retryAfter,
                                      String responseBody, RequestSummary requestSummary,
                                      int attempts, Throwable cause) {
        super(message, errorCode, httpStatus, requestId, errorId,
                retryAfter, responseBody, requestSummary, attempts, cause);
    }
}

File: packages/csharp/src/Errors/InvalidOrderStateException.cs
using System;

namespace Tesote.Sdk.Errors;

/// <summary>409 INVALID_ORDER_STATE — order cannot transition from current state.</summary>
public sealed class InvalidOrderStateException : ApiException
{
    /// <summary>Construct with the full required-field set.</summary>
    public InvalidOrderStateException(
        string? message, string? errorCode, int httpStatus,
        string? requestId, string? errorId, int? retryAfter,
        string? responseBody, RequestSummary? requestSummary,
        int attempts, Exception? cause)
        : base(message, errorCode, httpStatus, requestId, errorId,
               retryAfter, responseBody, requestSummary, attempts, cause)
    {
    }
}
